internal/provider/resource_packer_channel_assignment.go

```diff
@@ -0,0 +1,458 @@
+
+	channel, err := clients.GetPackerChannelBySlugFromList(ctx, client, loc, bucketName, channelName)
+	if err != nil {
+		return diag.FromErr(err)
+	} else if channel == nil {
+		return diag.Diagnostics{diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  fmt.Sprintf("HCP Packer channel with (channel_name %q) (bucket_name %q) (project_id %q) not found.", channelName, bucketName, loc.ProjectID),
+		}}
+	} else if channel.Managed {
+		return diag.Diagnostics{diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  fmt.Sprintf("HCP Packer channel with (channel_name %q) (bucket_name %q) (project_id %q) is managed by HCP Packer and cannot have an iteration assigned by Terraform.", channelName, bucketName, loc.ProjectID),
+		}}
+	} else if iteration := channel.Iteration; iteration != nil && (iteration.IncrementalVersion > 0 || iteration.ID != "" || iteration.Fingerprint != "") {
+		return diag.Diagnostics{diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  fmt.Sprintf("HCP Packer channel with (channel_name %q) (bucket_name %q) (project_id %q) already has an assigned iteration.", channelName, bucketName, loc.ProjectID),
+			Detail:   "To adopt this resource into Terraform, use `terraform import`, or remove the channel's assigned iteration using the HCP Packer GUI/API.",
+		}}
+	}
+
+	iterID := d.Get("iteration_id").(string)
+	if iterID == "none" {
+		iterID = ""
+	}
+
+	iterFingerprint := d.Get("iteration_fingerprint").(string)
+	if iterFingerprint == "none" {
+		iterFingerprint = ""
+	}
+
+	updatedChannel, err := clients.UpdateBucketChannel(ctx, client, loc, bucketName, channelName,
+		&packermodels.HashicorpCloudPackerIteration{
+			IncrementalVersion: int32(d.Get("iteration_version").(int)),
+			ID:                 iterID,
+			Fingerprint:        iterFingerprint,
+		}, nil,
+	)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(updatedChannel.ID)
+
+	if err := setPackerChannelAssignmentIterationData(d, updatedChannel.Iteration); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourcePackerChannelAssignmentUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*clients.Client)
+	projectID, err := GetProjectID(d.Get("project_id").(string), client.Config.ProjectID)
+	if err != nil {
+		return diag.Errorf("unable to retrieve project ID: %v", err)
+	}
+
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: client.Config.OrganizationID,
+		ProjectID:      projectID,
+	}
+	if err := setLocationData(d, loc); err != nil {
+		return diag.FromErr(err)
+	}
+
+	bucketName := d.Get("bucket_name").(string)
+	channelName := d.Get("channel_name").(string)
+
+	iteration := &packermodels.HashicorpCloudPackerIteration{}
+	assignmentHasChanges := d.HasChanges("iteration_version", "iteration_id", "iteration_fingerprint")
+	if !assignmentHasChanges || d.HasChange("iteration_version") {
+		iteration.IncrementalVersion = int32(d.Get("iteration_version").(int))
```

Comment:
I'm a little confused on what the expected value of iteration_version is meant to be here?

Reply:
`iteration_version` contains the incremental version. Terraform stores this value as an `int`, but the iteration struct used throughout the code expects an `int32`

Reply:
Yup, you got it